From 91f065d9b4a77ff6ec7c103974c27399dc0c989c Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@gmail.com>
Date: Fri, 11 Jun 2021 00:49:23 -0500
Subject: [PATCH 1/2] Don't build make depends if the package is already built

If we are attempting to install/build a package and that package is
already built, we don't need to consider any make depends of said
package.

Closes #29
---
 kiss | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kiss b/kiss
index f492259..b1182a2 100755
--- a/kiss
+++ b/kiss
@@ -369,8 +369,8 @@ pkg_depends() {
         [ "$3" ] && [ -z "$2" ] && (pkg_list "$1" >/dev/null) && return
 
         # Recurse through the dependencies of the child packages.
-        while read -r dep _ || [ "$dep" ]; do
-            [ "${dep##\#*}" ] && pkg_depends "$dep" '' "$3"
+        while read -r dep make || [ "$dep" ]; do
+            { [ "make" != "$make" ] || ! pkg_cache "$1"; } && [ "${dep##\#*}" ] && pkg_depends "$dep" '' "$3"
         done 2>/dev/null < "$(pkg_find "$1")/depends" ||:
 
         # After child dependencies are added to the list,
-- 
2.31.1

